File: CPP/src/Bison.cpp
#include <iostream>
#include "../include/Bison.hpp"

/*
Constructor yang menerima parameter position, name, meat, dan milk
Parameter diturunkan dari MeatProducing dan MilkProducing.
*/
Bison::Bison(pair<int,int> _position, string _name, Land* _landPos){
    initObject(BISON, 'B');
    initializeFarmAnimal(_position, _name, _landPos);
    initializeMeatProducing(new BisonMeat(HARGA_DAGING_BISON, getName()));
    initializeMilkProducing(NULL);
}

void Bison::produceMeat(Player& _player){
    if (meat != NULL){
        _player.addInventory(new BisonMeat(HARGA_DAGING_BISON, getName()));
        delete meat;
    }
}

void Bison::produceMilk(Player& _player){
    if (milk != NULL){
        _player.addInventory(new BisonMilk(HARGA_SUSU_BISON, getName()));
        delete milk;
    }
}

void Bison::eat(){
    FarmAnimal::eat();
    setIsProductProduced(false);
    initializeMilkProducing(new BisonMilk(HARGA_SUSU_BISON, getName()));
}

void Bison::talk(){
    cout << name << ": Mooo Ahahahah!" << endl;
}

void Bison::interact(Player* _p){
    if (!getIsProductProduced()){
        produceMilk(*_p);
        setIsProductProduced(true);
    }
}

File: Java/src/engisubs/gameobject/farmanimal/Bison.java
package engisubs.gameobject.farmanimal;

import java.util.*;
import engisubs.gameobject.*;
import engisubs.gameobject.cell.land.Land;
import engisubs.gameobject.product.farmproduct.meat.*;
import engisubs.gameobject.product.farmproduct.milk.*;

public class Bison extends FarmAnimal implements MeatProducing, MilkProducing{
    public Bison(Map <String, Integer> _position, String _name, Land _landPos){
        super.killable = true;
        initializeFarmAnimal(_position, _name, _landPos);
        initGameObject(GameObjectType.BISON, 'B');
    }

    public void produceMeat(Player _p){
        _p.addInventory(new BisonMeat(100, name));
    }

    public void produceMilk(Player _p){
        _p.addInventory(new BisonMilk(20, name));
    }

    public void talk(){
        System.out.println(name + ": Mwooahahaha!");
    }

    public void interact(Player _p){
        if (!getIsProductProduced()){
            produceMilk(_p);
            isProductProduced = true;
        }
    }

}